actionpack/lib/action_dispatch/http/uri.rb

```diff
@@ -0,0 +1,155 @@
+# frozen_string_literal: true
+
+require "active_support/core_ext/module/attribute_accessors"
+require "uri/generic"
+
+module ActionDispatch
+  module Http
+    class URI
+      IP_HOST_REGEXP = /\d{1,3}\.\d{1,3}\.\d{1,3}\.\d{1,3}$/
```

Comment:
This regexp also matches `example.com.1.2.3.4`
```suggestion
      IP_HOST_REGEXP = /\A\d{1,3}\.\d{1,3}\.\d{1,3}\.\d{1,3}\z/
```